service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/DumpAzureDatabaseStep.java

```diff
@@ -111,8 +113,16 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         destinationContainer.getStorageContainerName(),
         blobFileName);
 
+    var sourceAzureContext =
+        azureCloudContextService
+            .getAzureCloudContext(sourceDatabase.getWorkspaceId())
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "Cloud context for the workspace containing the source database could not be found."));
+
     this.azureDatabaseUtilsRunner.pgDumpDatabase(
-        workingMap.get(AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
+        sourceAzureContext,
```

Comment:
Interesting that this wasn't already what was in the working map (should we have source and destination contexts available there?). And is there any reason why we don't also need an equivalent change in the restore step?

Reply:
Same question regarding if this change also needs to be added to the restore step

Reply:
I didn't check exhaustively, but I think the source context isn't in the working map because this pgDump step is the only step (at the moment) that needs to run azureDatabaseUtils pods in the source workspace context.

We don't need to change anything in the restore step because the azureDatabaseUtils pod that it deploys runs in the destination workspace context already.